python/pyiceberg/expressions/visitors.py

```diff
@@ -986,3 +989,263 @@ def expression_to_plain_format(
+        field = term.ref().field
+        field_id = field.field_id
+
+        if self._contains_nulls_only(field_id) or self._contains_nans_only(field_id):
+            return ROWS_CANNOT_MATCH
+
+        if not isinstance(field.field_type, PrimitiveType):
+            raise ValueError(f"Expected PrimitiveType: {field.field_type}")
+
+        if field_id in self.lower_bounds:
+            lower = from_bytes(field.field_type, self.lower_bounds[field_id])
+
+            if self._is_nan(lower):
+                # NaN indicates unreliable bounds. See the InclusiveMetricsEvaluator docs for more.
+                return ROWS_MIGHT_MATCH
+
+            if lower > literal.value:
+                return ROWS_CANNOT_MATCH
+
+        return ROWS_MIGHT_MATCH
+
+    def visit_greater_than(self, term: BoundTerm[L], literal: Literal[L]) -> bool:
+        field = term.ref().field
+        field_id = field.field_id
+
+        if self._contains_nulls_only(field_id) or self._contains_nans_only(field_id):
+            return ROWS_CANNOT_MATCH
+
+        if not isinstance(field.field_type, PrimitiveType):
+            raise ValueError(f"Expected PrimitiveType: {field.field_type}")
+
+        if field_id in self.upper_bounds and from_bytes(field.field_type, self.upper_bounds[field_id]) <= literal.value:
+            return ROWS_CANNOT_MATCH
+
+        return ROWS_MIGHT_MATCH
+
+    def visit_greater_than_or_equal(self, term: BoundTerm[L], literal: Literal[L]) -> bool:
+        field = term.ref().field
+        field_id = field.field_id
+
+        if self._contains_nulls_only(field_id) or self._contains_nans_only(field_id):
+            return ROWS_CANNOT_MATCH
+
+        if not isinstance(field.field_type, PrimitiveType):
+            raise ValueError(f"Expected PrimitiveType: {field.field_type}")
+
+        if field_id in self.upper_bounds and from_bytes(field.field_type, self.upper_bounds[field_id]) < literal.value:
+            return ROWS_CANNOT_MATCH
+
+        return ROWS_MIGHT_MATCH
+
+    def visit_equal(self, term: BoundTerm[L], literal: Literal[L]) -> bool:
+        field = term.ref().field
+        field_id = field.field_id
+
+        if self._contains_nulls_only(field_id) or self._contains_nans_only(field_id):
+            return ROWS_CANNOT_MATCH
+
+        if not isinstance(field.field_type, PrimitiveType):
+            raise ValueError(f"Expected PrimitiveType: {field.field_type}")
+
+        if field_id in self.lower_bounds:
+            lower = from_bytes(field.field_type, self.lower_bounds[field_id])
+
+            if self._is_nan(lower):
+                # NaN indicates unreliable bounds. See the InclusiveMetricsEvaluator docs for more.
+                return ROWS_MIGHT_MATCH
+
+            if lower > literal.value:
+                return ROWS_CANNOT_MATCH
+
+        if field_id in self.upper_bounds and from_bytes(field.field_type, self.upper_bounds[field_id]) < literal.value:
+            return ROWS_CANNOT_MATCH
```

Comment:
I think we want to match the structure of the lower bounds check and include an `isnan` check here, too.

Reply:
Got it, I misread the other comment, thanks!